src/main/java/com/autotune/analyzer/recommendations/engine/DurationBasedRecommendationEngine.java

```diff
@@ -241,6 +246,53 @@ public HashMap<String, Recommendation> getRecommendations(ContainerData containe
         return resultRecommendation;
     }
 
+    @Override
+    public boolean checKIfMinDataAvailable(ContainerData containerData) {
+        // Initiate to the first sub category available
+        DurationBasedRecommendationSubCategory categoryToConsider = (DurationBasedRecommendationSubCategory) this.category.getRecommendationSubCategories()[0];
+        // Loop over categories to set the least category
+        for (RecommendationSubCategory recommendationSubCategory : this.category.getRecommendationSubCategories()) {
+            DurationBasedRecommendationSubCategory durationBasedRecommendationSubCategory = (DurationBasedRecommendationSubCategory) recommendationSubCategory;
+            if (durationBasedRecommendationSubCategory.getDuration() < categoryToConsider.getDuration()) {
+                categoryToConsider = durationBasedRecommendationSubCategory;
+            }
+        }
+        // Set bounds to check if we get minimum requirement satisfied
+        double lowerBound = categoryToConsider.getGetDurationLowerBound();
+        double sum = 0.0;
+        // Loop over the data to check if there is
+        for (IntervalResults intervalResults: containerData.getResults().values()) {
+            sum = sum + intervalResults.getDurationInMinutes();
+            // We don't consider upper bound to check if sum is in-between as we may over shoot and end-up resulting false
+            if (sum >= lowerBound)
+                return true;
+        }
+        return false;
+    }
+
+    private static Timestamp getMonitoringStartDate(HashMap<Timestamp, IntervalResults> resultsHashMap,
```

Comment:
This should be `getMonitoringStartTime` following the nomenclature

Reply:
Thanks @dinogun for the review, Made changes accordingly